src/execution/sql/runtime_types.cpp

```diff
@@ -1,118 +1,644 @@
 #include "execution/sql/runtime_types.h"
-#include <date/date.h>
 
 #include <string>
 
 #include "common/exception.h"
-#include "util/time_util.h"
+#include "spdlog/fmt/fmt.h"
 
 namespace terrier::execution::sql {
 
 namespace {
 
-// The below Julian date conversions are taken from Postgres.
+constexpr int64_t K_MONTHS_PER_YEAR = 12;
+constexpr int32_t K_DAYS_PER_MONTH[2][12] = {{31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31},
+                                             {31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31}};
+constexpr int64_t K_HOURS_PER_DAY = 24;
+constexpr int64_t K_MINUTES_PER_HOUR = 60;
+constexpr int64_t K_SECONDS_PER_MINUTE = 60;
+constexpr int64_t K_MILLISECONDS_PER_SECOND = 1000;
+constexpr int64_t K_MICROSECONDS_PER_MILLISECOND = 1000;
 
-bool IsValidJulianDate(int32_t year, uint32_t month, uint32_t day) {
-  return year <= 9999 && month >= 1 && month <= 12 && day >= 1 && day <= 31;
+constexpr char K_TIMESTAMP_SUFFIX[] = "::timestamp";
```

Comment:
@pmenon why a `constexpr char[]` and not a `constexpr std::string_view` since we're calling `strlen` on it every time (which in theory should be `constexpr` as well but the STL doesn't have a `constexpr` `strlen`)?

Reply:
`strlen` should be optimized out. I originally suggested `sizeof` to be sure.

I'm okay with this change. In fact, you can even remove the new function with [`std::string_view::ends_with()`](https://en.cppreference.com/w/cpp/string/basic_string_view/ends_with) if we switch to C++20 ... which isn't a bad idea.

Reply:
Should I replace K_TIMESTAMP_SUFFIX with a string_view and reface the arguments in the call to EndsWith as K_TIMESTAMP_SUFFIX.data() and K_TIMESTAMP_SUFFIX.size() instead?

Reply:
Sounds reasonable.